pkg/storage/common/common.go

```diff
@@ -361,6 +365,19 @@ func RemoveManifestDescByReference(index *ispec.Index, reference string, detectC
 		return ispec.Descriptor{}, zerr.ErrManifestNotFound
 	}
 
+	// In case of delete by digest we remove the manifest right away from storage
```

Comment:
What happens if tag1 -> digest and tag2 -> digest, if we delete by digest immediately, tag1 and tag2 are dangling?

Reply:
No, they are all deleted